packages/google_sign_in/google_sign_in_web/lib/google_sign_in_web.dart

```diff
@@ -83,8 +90,11 @@ class GoogleSignInPlugin extends GoogleSignInPlatform {
   }
 
   @override
-  Future<void> initWithParams(SignInInitParameters params) async {
-    final String? appClientId = params.clientId ?? _autoDetectedClientId;
+  Future<void> initWithParams(
+    SignInInitParameters params, {
+    @visibleForTesting GisSdkClient? overrideClient,
```

Comment:
This is cool! I didn't know you could annotate a single parameter like this.

Reply:
Yes! You even get an analyzer warning if you use from another package!

<img width="630" alt="Screenshot 2023-02-03 at 2 13 25 PM" src="https://user-images.githubusercontent.com/1255594/216720898-a3b7292b-2c66-4350-ac0d-ee371e76a130.png">